### PR TITLE
Add Updated and Status columns to Snippets listing and other improvements 

### PR DESCRIPTION
**Pull request by laymonage**
- Add "Updated" and "Status" columns to generic `IndexView`
- Add "Instances" column to snippets model index view
- Allow sorting columns
- Styles improvements to match new UI

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/admin/views/generic/models.py [hunk 1]
```diff
@@ -186,26 +234,51 @@ def search_queryset(self, queryset):
...
+            if draftstate_enabled and obj.latest_revision:
+                return obj.latest_revision.object_str
+            return str(obj)
+
+        return column_class(
+            "name",
+            label=gettext_lazy("Name"),
+            accessor=title_accessor,
+            get_url=self.get_edit_url,
+        )
```
Comment (laymonage) [inline wagtail/admin/views/generic/models.py:250]: Decided not to handle the ordering of the "Name" column in this PR. I think it's better to revisit this when we design the SnippetViewSet API.

File: wagtail/admin/views/generic/models.py [hunk 2]
```diff
@@ -186,26 +234,51 @@ def search_queryset(self, queryset):
...
+
+    def _get_updated_at_column(self, column_class=DateColumn):
+        return column_class("_updated_at", label=_("Updated"), sort_key="_updated_at")
```
Comment (kaedroho) [inline wagtail/admin/views/generic/models.py:253]: Issue: I'm unable to reverse the ordering of the "Updated" column on snippets. Clicking it once adds ``?ordering=_updated_at`` and sorts the listing by updated, but clicking it again does not change it to ``?ordering=-_updated_at``.

**Review by kaedroho — CHANGES_REQUESTED** [review]
The back end code all looks good to me! Just an issue with reversing the ordering of the "Updated" column.

File: wagtail/admin/views/generic/models.py (hunk 2 above)
Comment (laymonage) [reply wagtail/admin/views/generic/models.py:259]: Oops, sorry! Made a mistake when handling null values. Should be fixed now and I've added the tests to ensure it. If you reviewed each commit, that means you just need to re-review the two last commits. Thanks!

**Review by kaedroho — APPROVED** [review]
All looks good to me, nice work!

**Pull request merged**